endpoint_info.go

```diff
@@ -162,12 +162,16 @@ func (ep *endpoint) Interface() driverapi.InterfaceInfo {
 }
 
 func (epi *endpointInterface) SetMacAddress(mac net.HardwareAddr) error {
-	if epi.mac != nil {
-		return types.ForbiddenErrorf("endpoint interface MAC address present (%s). Cannot be modified with %s.", epi.mac, mac)
-	}
 	if mac == nil {
 		return types.BadRequestErrorf("tried to set nil MAC address to endpoint interface")
 	}
+	if epi.mac != nil {
```

Comment:
The current contract between libnetwork and network driver is in fact that network driver will set only the information that was missing. `overlay` and `remote` drivers are following this.

If the MAC and addresses passed during call to `CreateEndpoint` are set in the bridge `endpoint` structure, it should then be possible before returning from the call to set only the fields which are missing.

Can you point to an example, that would help understand the issue.